pubsub/cloud-client/quickstart/sub.py

```diff
@@ -34,20 +33,22 @@ def sub(project_id, subscription_name):
         project_id, subscription_name)
 
     def callback(message):
-        print('Received message {} of message ID {}'.format(
+        print('Received message {} of message ID {}\n'.format(
             message, message.message_id))
         # Acknowledge the message. Unack'ed messages will be redelivered.
```

Comment:
(optional nit)

```
# Unacknowledged messages will be redelivered when a client subscribes.
```

I previously intuited that this client would just keep getting the same message if the message was not ack'd which is not the case. Only if **this** client reconnects will the message get redelivered to this client, which can happen if the loop exits because of an exception, etc...

Reply:
Thanks for the comment Gus. That's something that I have never thought much of. I think Pub/Sub load-balances if there are multiple subscribers, but only if the load is large (like >10Mb). I remember trying with just three messages and they would keep getting delivered to the same subscriber client.